app/model/sdl/Abstract/Model.js

```diff
@@ -200,48 +200,29 @@ SDL.SDLModel = Em.Object.extend({
   onFileRemoved: function(params) {
 
     var result = false;
+    var app_model = SDL.SDLController.getApplicationModel(params.appID);
+    var is_image_type = ['GRAPHIC_PNG', 'GRAPHIC_BMP', 'GRAPHIC_JPEG'].includes(params.fileType);
 
-    if ((
-      params.fileType === 'GRAPHIC_PNG' || params.fileType === 'GRAPHIC_BMP' ||
-      params.fileType === 'GRAPHIC_JPEG') &&
-      SDL.SDLController.getApplicationModel(params.appID)) {
-      result = SDL.SDLController.getApplicationModel(params.appID).
-                   onImageRemoved(params.fileName);
-
-      if (SDL.SDLController.getApplicationModel(params.appID).appIcon.
-              indexOf(params.fileName) != -1 &&
-        params.fileName.length ==
-        SDL.SDLController.getApplicationModel(params.appID).appIcon.length) {
-        SDL.SDLController.getApplicationModel(params.appID).
-            set('appIcon', SDL.SDLModel.data.defaultListOfIcons.app);
+    if (is_image_type && app_model) {
+      result = app_model.onImageRemoved(params.fileName);
+
+      if (app_model.appIcon.indexOf(params.fileName) != -1 &&
+        params.fileName.length == app_model.appIcon.length) {
+        app_model.set('appIcon', SDL.SDLModel.data.defaultListOfIcons.app);
       }
 
-      if (SDL.SDLController.getApplicationModel(params.appID
-        ).constantTBTParams) {
-
-        if (SDL.SDLController.getApplicationModel(params.appID
-          ).constantTBTParams.turnIcon &&
-          SDL.SDLController.getApplicationModel(params.appID).constantTBTParams.
-              turnIcon.value.indexOf(params.fileName) != -1 &&
-          params.fileName.length == SDL.SDLController.getApplicationModel(
-            params.appID
-          ).constantTBTParams.turnIcon.value.length) {
-          SDL.SDLController.getApplicationModel(params.appID
-          ).constantTBTParams.turnIcon.value            =
-              SDL.SDLModel.data.defaultListOfIcons.command;
+      if (app_model.constantTBTParams) {
+        if (app_model.constantTBTParams.turnIcon &&
+            app_model.constantTBTParams.turnIcon.value.indexOf(params.fileName) != -1 &&
+          params.fileName.length == app_model.constantTBTParams.turnIcon.value.length) {
```

Comment:
Wouldn't you get the same result if you replaced these conditions with 
```suggestion
            app_model.constantTBTParams.turnIcon.value === params.fileName) {
```

Reply:
@ShobhitAd agree, you suggestion is much obvious. Fixed in https://github.com/smartdevicelink/sdl_hmi/pull/332/commits/122e39649b6c14a42f13bac8cd4a6412d75fb591